src/hatch/template/files_feature_ci.py

```diff
@@ -38,9 +38,24 @@ class CommandLinePackage(File):
       with:
         python-version: ${{ matrix.python-version }}
 
+    - uses: actions/cache@v3
+      id: cache
+      if: runner.os != 'Windows'
+      with:
+        path: ${{ env.pythonLocation }}
+        key: ${{ runner.os }}-python-${{ env.pythonLocation }}-${{ hashFiles('pyproject.toml') }}-test-v02
```

Comment:
I don't think this is a good enough cache key — it may break when GitHub updates Python in their images. I'd go for a hash of the full version information instead of just path.